libs/utils/android/workloads/vellamo.py

```diff
@@ -0,0 +1,167 @@
+
+    # Setup logger
+    logger = logging.getLogger('Vellamo')
+    logger.setLevel(logging.DEBUG)
+
+    def __init__(self, test_env):
+        super(Vellamo, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'Vellamo')
+
+    def wload_run(self, exp_dir, test_name, collect=''):
+        """
+        Run single Vellamo workload. Returns a collection of results.
+
+        :param exp_dir: Path to experiment directory where to store results.
+        :type exp_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+        :type collect: list(str)
+        """
+
+        if test_name.upper() not in VELLAMO_TESTS:
+            raise ValueError('Vellamo workload [%s] not supported', test_name)
+
+        # Set parameter depending on test
+        self.logger.debug('Start Vellamo Benchmark [%s]', test_name)
+        test_cmd = ''
+        sleep_time = 0
+        if test_name.upper() == 'BROWSER':
+            test_cmd = 'input tap 970 740'
+            sleep_time = 3.5
+        elif test_name.upper() == 'METAL':
+            test_cmd  = 'input tap 970 1275'
+            sleep_time = 1
+        elif test_name.upper() == 'MULTI':
+            test_cmd = 'input tap 970 1555'
+            sleep_time = 3.5
+
+        nrg_data, nrg_file = None, None
+
+        # Unlock device screen (assume no password required)
+        self.target.execute('input keyevent 82')
+        # Start Vellamo on the target device
+        self.target.execute('pm clear {}'.format(self.package))
+        self.target.execute('am start -W -n {}/{}'.format(self.package,
+                                                          self.activity))
+        # Wait few seconds to make sure the app is loaded
+        sleep(1)
+
+        # Accept EULA
+        self.target.execute('input tap 864 1646')
+        sleep(1)
+        # Click Let's Roll
+        self.target.execute('input tap 460 1400')
+        sleep(1)
+        # Skip Arrow
+        self.target.execute('input tap 500 1500')
+
+        # Start measuring energy
+        if 'energy' in collect and self.te.emeter:
+            self.te.emeter.reset()
+
+        # Run Workload
+        self.target.execute(test_cmd)
+
+        # Skip instructions
+        self.target.execute('input swipe 900 1000 200 1000 100')
+        self.target.execute('input swipe 900 1000 200 1000 100')
+        self.target.execute('input swipe 900 1000 200 1000 100')
+        self.logger.debug("Vellamo - {} started!".format(test_name.upper()))
+
+        # Wait for completion
+        sleep(sleep_time*60)
+
+        # Stop measuring energy
+        if 'energy' in collect and self.te.emeter:
+            nrg_data, nrg_file = self.te.emeter.report(exp_dir)
+            self.logger.info('%s - Estimated energy: %7.3f',
+                             test_name,
+                             float(nrg_data['BAT']))
```

Comment:
Do you think it could be sufficient, at least for the time being, to expose the channel name as a run() method parameter?!? At the end the EMeter is configured by the caller... thus:
1. if we specify a channel: that means we want to collect energy and we know also from which channel.
```python
def wload_run(self, exp_dir, test_name, collect='', emeter_channel=None):
    #....
    if 'energy' in collect and self.te.emeter and emeter_channel:
        #....
```